src/service/fixed-layer.js

```diff
@@ -449,15 +450,36 @@ export class FixedLayer {
           // We shouldn't have too many of `fixed` elements.
           break;
         }
-        this.setupElement_(elements[j], fixedSelector, 'fixed');
+        const el = elements[j];
+        this.removeStyleAttributeIfNecessary_(el);
+        this.setupElement_(el, fixedSelector, 'fixed');
       }
     }
     for (let i = 0; i < stickySelectors.length; i++) {
       const stickySelector = stickySelectors[i];
       const elements = this.ampdoc.getRootNode().querySelectorAll(
           stickySelector);
       for (let j = 0; j < elements.length; j++) {
-        this.setupElement_(elements[j], stickySelector, 'sticky');
+        const element = elements[j];
+        this.removeStyleAttributeIfNecessary_(element);
+        this.setupElement_(element, stickySelector, 'sticky');
+      }
+    }
+  }
+
+  /**
+   * If the given element has a `style` attribute, remove it and throw a user
+   * error. This is done since FixedLayer's implementation currently assumes
+   * that publisher-authored inline styles are impossible.
+   * @param {!Element} element
+   * @private
+   */
+  removeStyleAttributeIfNecessary_(element) {
+    if (isExperimentOn(this.ampdoc.win, 'inline-styles')) {
+      if (element.hasAttribute('style')) {
```

Comment:
All styles? Not just `position`?

Reply:
Yes, this is the first phase of implementing the changes you proposed, the minimal changes required to unblock pinterest. In their example as per comment https://github.com/ampproject/amphtml/issues/11881#issuecomment-351826881, they want to inline the css in the example they provided, which currently just uses 'absolute' positioning. So for now, we don't want to allow any styling in fixed elements.

Reply:
I think Dima is suggesting that we could just check for `position:` rules in the style attribute instead of banning `style` outright. 

@dvoytenko Reasonable request but is it okay if we address it in a follow-up?

Reply:
Fixed.